src/client.ts

```diff
@@ -452,6 +477,84 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
     return this._client.getFeatureVariableDouble(feature, variable, user.id, user.attributes);
   }
 
+  /**
+   * Returns value for the given json variable attached to the given feature
+   * flag
+   * @param {string} feature
+   * @param {string} variable
+   * @param {string} [overrideUserId]
+   * @param {optimizely.UserAttributes} [overrideAttributes]
+   * @returns {(string | null)}
+   * @memberof OptimizelyReactSDKClient
+   */
+  public getFeatureVariableJson(
+    feature: string,
+    variable: string,
+    overrideUserId?: string,
+    overrideAttributes?: optimizely.UserAttributes
+  ): unknown {
+    const user = this.getUserContextWithOverrides(overrideUserId, overrideAttributes);
+    if (user.id === null) {
+      return null
+    }
+    return this._client.getFeatureVariableJson(
+      feature,
+      variable,
+      user.id,
+      user.attributes,
+    );
+  }
+
+  /**
+   * Returns dynamically-typed value of the variable attached to the given
+   * feature flag. Returns null if the feature key or variable key is invalid.
+   * @param {string} featureKey
+   * @param {string} variableKey
+   * @param {string} [overrideUserId]
+   * @param {optimizely.UserAttributes} [overrideAttributes]
+   */
+  getFeatureVariable(
+      featureKey: string,
+      variableKey: string,
+      overrideUserId: string,
+      overrideAttributes?: optimizely.UserAttributes
+  ): unknown {
+    const user = this.getUserContextWithOverrides(overrideUserId, overrideAttributes);
+    if (user.id === null) {
+      return null
+    }
+    return this._client.getFeatureVariable(
+        featureKey,
+        variableKey,
+        user.id,
+        user.attributes,
+    );
+  }
+
+  /**
+   * Returns values for all the variables attached to the given feature flag
+   * @param {string} featureKey
+   * @param {string} overrideUserId
+   * @param {optimizely.UserAttributes} [overrideAttributes]
+   * @returns {(object | null)}
+   * @memberof OptimizelyReactSDKClient
+   */
+  getAllFeatureVariables(
```

Comment:
@mjc1283 Want your take on it. We are adding this as the wrapper to the corresponding javascript sdk method. There is another public function `getFeatureVariables` as part of the same client object. This is going to be confusing for the end user eventually. There are two functions doing the same thing with different name. The only difference is that `getFeatureVariables` sends separate notification for each variable whereas the new `getAllFeatureVariables` sends one single notification containing all variables. Do you want us to do anything about it as part of this PR or you want to revisit later.?

Reply:
We should deprecate `getFeatureVariables` and remove it in the next major version. The hook that uses `getFeatureVariables` should be updated to use `getAllFeatureVariables`.

In this PR I recommend the following:
- Add a deprecated comment to `getFeatureVariables`
- Add a deprecated log message at WARNING level when `getFeatureVariables` is called
- Add a deprecated note to the README documentation for `getFeatureVariables`

Reply:
Actually, let's skip the log warning. We don't want to be logging that every time the feature hook is used.

Reply:
Deprecated and added snapshot of Readme in Description